src/pybind/mgr/dashboard/frontend/src/app/ceph/block/rbd-form/rbd-form.component.ts

```diff
@@ -217,15 +216,9 @@ export class RbdFormComponent extends CdForm implements OnInit {
     this.rbdForm.get('obj_size').disable();
     this.rbdForm.get('stripingUnit').disable();
     this.rbdForm.get('stripingCount').disable();
-
-    /* RBD Image Format v1 */
-    this.rbdImage.subscribe((image: RbdModel) => {
-      if (image.image_format === RBDImageFormat.V1) {
-        this.rbdForm.get('deep-flatten').disable();
-        this.rbdForm.get('layering').disable();
-        this.rbdForm.get('exclusive-lock').disable();
-      }
-    });
+    this.rbdForm.get('deep-flatten').disable();
+    this.rbdForm.get('layering').disable();
+    this.rbdForm.get('exclusive-lock').disable();
```

Comment:
This doesn't seem right: `exclusive-lock` feature can be disabled after the features that depend on it  (`object-map`/`fast-diff`, `journaling`) are disabled and can be re-enabled later.

Reply:
Currently we have this behavior, where `object-map` can only be enabled if `exclusive-lock` is enabled. Is this correct? @idryomov 

[Screencast from 2023-08-28 15-58-37.webm](https://github.com/ceph/ceph/assets/23611106/e9d89851-2d01-48d4-94fc-869ef79ffeed)

Reply:
> Currently we have this behavior, where `object-map` can only be enabled if `exclusive-lock` is enabled. Is this correct?

Yes -- `object-map` depends on `exclusive-lock`.

However, there is room for improvement.  "Interlocked" means that for all intents and purposes, `object-map` and `fast-diff` are the same feature.  They can't really be enabled or disabled separately.  So when `exclusive-lock` is checked, I would suggest making both `object-map` and `fast-diff` check boxes available and if the user checks either of them, check the other automatically.

Reply:
Yes that's the behavior if exclusive-lock is enabled `object-map` gets is available & once `object-map` is enabled , `fast-diff` is available to be enabled, its a chain of `exclusive-lock->object-map->fast-diff`.
Attaching a screen cast  again to show it better way.
[Screencast from 2023-08-29 15-07-14.webm](https://github.com/ceph/ceph/assets/23611106/9ba92560-2fec-4ed7-8312-ac88ccf491b4)

Reply:
Yes the `object-map` gets unchecked too if `fast-diff` is unchecked.
[Screencast from 2023-08-29 16-34-07.webm](https://github.com/ceph/ceph/assets/23611106/15d450bf-aee4-48c9-86aa-5804fa5068b1)